### PR TITLE
Format the world

### PR DESCRIPTION
**Pull request by Mark-Simulacrum**
This PR modifies the formatting infrastructure a bit in the first commit (to enable the forgotten 2018 edition), as well as removes most directories from the ignore list in rustfmt.toml. It then follows that up with the second commit which runs `x.py fmt` and formats the entire repository.

We continue to not format the test directory (`src/test`) because of interactions with pretty printing and, in part, because re-blessing all of those files is somewhat harder to review, so is best suited for a follow up PR in my opinion.

----

Update: **Instructions for resolving merge conflicts** can be found in [the blog post on Inside Rust](https://blog.rust-lang.org/inside-rust/2019/12/23/formatting-the-compiler.html). —mbrubeck

### CODE REVIEW

**Comment by Centril** [comment]
Let's do this! 🚀 

r? @Centril @bors p=1337 rollup=never

File: src/librustc_mir/borrow_check/constraint_generation.rs [hunk 1]
```diff
@@ -6,16 +6,12 @@ use rustc::mir::{
...
 
 use crate::borrow_check::{
-    borrow_set::BorrowSet,
-    location::LocationTable,
-    nll::ToRegionVid,
-    facts::AllFacts,
-    region_infer::values::LivenessValues,
-    places_conflict,
+    borrow_set::BorrowSet, facts::AllFacts, location::LocationTable, nll::ToRegionVid,
+    places_conflict, region_infer::values::LivenessValues,
```
[the remark this answers is not in the document]
Comment (Mark-Simulacrum) [reply src/librustc_mir/borrow_check/constraint_generation.rs]: I consider this a failing of the import style -- I would prefer that we not have multi-line imports at all, as they make e.g. grepping for uses much harder (and lead to potential problems like this).

**Comment by Mark-Simulacrum** [comment]
@matthewjasper I agree that both of those that you noted aren't great, but I think if we block this until we're happy with the specific output it's not going to land.

However, with the consideration that reverting merge_derives later is hard, but not applying it now is easy, I've enabled that option. I suspect that we'll want to shy towards defaults in the long run, but I would say that it's a rustfmt bug to merge them when it causes you to go *really* multiline.

Force-pushed tidy fixes and the above.

**Comment by Mark-Simulacrum** [comment]
@bors r=Centril

Added what should be a retry around the msys2 install hopefully.

**Comment by estebank** [comment]
@Mark-Simulacrum shouldn't that retry be done in a separate PR for easier rollback?

**Comment by Mark-Simulacrum** [comment]
Doesn't really matter -- the retry is likely to stick around -- it's temporary in the sense that it's not really the right fix: we want to mirror msys2 to s3, but haven't gotten around to figuring out quite how to do that yet.

Regardless a separate PR does not make revert (much) easier, especially for the ~2 line change that this commit is.

**Comment by lzutao** [comment]
And that's cause a bunch of merge conflicts.

**Comment by mark-i-m** [comment]
I'm really glad this happened, but a bit of warning would have been nice...

**Comment by estebank** [comment]
[The blog has an announcement about this PR merging](https://blog.rust-lang.org/inside-rust/2019/12/23/formatting-the-compiler.html), but it was published once it was _fait accompli_.

@Mark-Simulacrum recommends using this snippet:

```bash
# This should be the name of the remote for rust-lang/rust
git fetch upstream
# This rebases up to the bors merge right before formatting landed;
# it needs to be done manually.
git rebase -i 9b98af84c4aa66392236fff59c86da2130d46d46
# This rebases onto the formatting PR (given the previous command, only that).
# We tell git to resolve all conflicts in favor of your code (`-Xtheirs`),
# and the `--exec` command specifies that after each commit lands, it will be formatted.
# This command will fail if your PR has intermediary commits with syntax conflicts.
git rebase -i a916ac22b9f7f1f0f7aba0a41a789b3ecd765018 \
    --exec './x.py fmt && git add -u && git commit --amend` \
    # This exec is optional, and won't work if your intermediate commits don't build,
    # but it helps make sure that the formatting resolution didn't introduce any errors.
    # It's recommended to run it afterwards before pushing at least.
    --exec './x.py check' \
    -Xtheirs
```

**Comment by Mark-Simulacrum** [comment]
This was brought up a couple times at compiler team meetings and the long term plan has been to do it around the holidays; I agree that it would've been a good idea to publish something on internals or so a bit earlier.

**Comment by RalfJung** [comment]
What this PR did do [the Miri error printing code](https://github.com/Mark-Simulacrum/rust/blob/0f24ccd21d9f734a21daaf3566900127167556d1/src/librustc/mir/interpret/error.rs#L441) is quite the disaster. Previously, we consistently had a `write!` macro at the beginning of the first line after the pattern, now the style is all over the place depending on how long the error message is. Is there anything we can do about that?

**Comment by Mark-Simulacrum** [comment]
`#[rustfmt::skip]` should just work, and is the recommendation I believe for such cases. (It can be applied at module-level granularity, but I believe will not be inherited).

**Comment by RalfJung** [comment]
This basically means I have to add `#[rustfmt::skip]` to all but the most trivial `match`, as rustfmt tends to destroy all of them. The long ones just show that better. Also I might still want the code inside the match arms to be formatted, it's just the match itself that rustfmt does a poor job on.

IMO this is a serious deficiency in rustfmt. (In fact I think almost all of my frustration about rustfmt stems from how it handles matches.) Cc https://github.com/rust-lang/rustfmt/issues/3995

**Pull request merged**